src/sessionid.ts

```diff
@@ -45,9 +48,16 @@ export class SessionIdManager {
         if (this._windowId) {
             return this._windowId
         }
-        if (!this.persistence.disabled && sessionStore.is_supported()) {
-            return sessionStore.parse(this.window_id_storage_key)
+
+        // In queue id will be undefined on fresh window opens
+        const inQueueId = sessionStore.parse(this.last_window_id_storage_key)
+        if (!this.persistence.disabled && sessionStore.is_supported() && inQueueId) {
```

Comment:
I'm wondering if this logic should be called in the class constructor? Right now, I think there may be odd behavior if the customer is not capturing data frequently (e.g. they have autocapture off and they use the `startSessionRecording` function to only record specific things). Specifically, I think you could hit a case where a user opens a new tab. That tab never captures anything, so this function is never called. Then, the user refreshes the page, and as a result, the initial `window_id` is persisted.